doc/design/to_run_stmt.md

```diff
@@ -0,0 +1,178 @@
+# To Run Statement
+
+## The Problem
+
+SQLFlow extends the SQL syntax and can describe an end-to-end machine learning pipeline. Data transformation is an important part in the entire pipeline. Currently we have the following two options for data transformation:
+
+- SQL.
+- COLUMN clause for per data instance transform. The transform logic is saved into the model.
+
+But that's not enough. It's a common case that the transformation is not per data instance which COLUMN clause is not suitable for. And we may need write very complex SQL or even cannot use SQL to express the transfomration logic. Let's take `extract_features` from [tsfresh package](https://tsfresh.readthedocs.io/en/latest/api/tsfresh.feature_extraction.html#module-tsfresh.feature_extraction.extraction) for example. It will preprocess the time series data, calculate statistical values using various analysis functions and parameters, and then generate tens or hunderds of features. [Convert the time series data of sliding window to rows](https://github.com/sql-machine-learning/sqlflow/issues/2238) is also a typical scenario.
+
+We can express the data transform logic using Python, leverage many mature python packages and encapsulate it into a reusable function. And then we propose to add the `TO RUN` clause in SQLFlow to call it.
+
+## Syntax Extension
+
+```SQL
+SELECT * FROM source_table
+TO RUN a_data_scientist/maxcompute_functions:1.0/my_func.data_preprocessor.a_python_func
+WITH param_a = value_a,
+     param_b = value_b
+INTO result_table
```

Comment:
I feel that we don't need INTO clause here.  The existence of the INTO clause implies that the Python function will generate a table. How if it doesn't generate any table, or if it generates more than one tables?

Reply:
`INTO` clause is optional here. If we don't need output any data, we don't need the `INTO` suffix.